sklearn/tree/_splitter.pyx

```diff
@@ -1085,8 +1051,9 @@ cdef class BestSparseSplitter(BaseSparseSplitter):
                              &is_samples_sorted)
 
             # Sort the positive and negative parts of `Xf`
-            simultaneous_sort(Xf + start, samples + start, end_negative - start)
-            simultaneous_sort(Xf + start_positive, samples + start_positive, end - start_positive)
+            simultaneous_sort(&Xf[start], &samples[start], end_negative - start)
+            if start_positive < end:
+                simultaneous_sort(&Xf[start_positive], &samples[start_positive], end - start_positive)
```

Comment:
This fixes a bug on main where `start_positive == end`, which can lead to `samples[start_positive]` being out of bounds.

Reply:
Would it be possible to write a non-regression test to trigger this case?